### ISSUE TITLE
Implement Facility Locator error messages 

### ISSUE THREAD

**Issue by mmiddaugh** [department-of-veterans-affairs/va.gov-team#18351]
## User Story or Problem Statement

As a Veteran using a screenreader to search on Facility Locator, I need to hear errors announced relative to the field where the error occurs so I know how to resolve the error. 

- **❗️ Must** for if the feedback must be applied 

## Details

1. The error message itself is not specific. Screen reader users won't know what exactly is the problem if the problem is called "this" instead of saying "Fill in a city, state, or postal code."
2. Errors are tooltips instead of being persistent within the label of the field itself.
https://user-images.githubusercontent.com/14154792/103782570-9f251c00-5005-11eb-955d-8a1a45420f50.mov

Facility Locator is not currently using the [design system for form field errors](https://design.va.gov/patterns/form-feedback). Depending on the future of the form system (and whether it will soon be replaced) we may need to simply style our error messages, separate from implementing form system. 
- The form field and its corresponding <label> are wrapped in a container with a class name of usa-input-error. This will provide a thick border and padding to visual indicate an error message to users who might have difficulty perceiving contrast. It also changes the border of the input to red as a secondary indicator.
- An error message is placed between the label and the form field.
- The form field receives an aria-describedby attribute that references the id of the error message. (Error has context, such as "Error, select an item from the Facility type list")
- Prepending <span class="sr-only">Error</span> to error messages alerts screen screen readers clearly that an error exists.
- Error message should be persistent
- Error message should appear in the field itself (not as tool tip)

## Tasks
- [ ] Implement error messages consistent with design spec and accessibility guidance
- [ ] Add appropriate cypress/unit testing

## Acceptance Criteria
If a Veteran clicks "Search" before any required field is populated (City, state or zip, Facility type and Service type when Facility type = Community providers (in VA's network)
- [x] the empty field's border is styled to visually indicate an error (with border thickness, padding and color)
- [x] an error message is placed between the label and the form field.
- [x] a contextual error message is announced to screenreader users
- [x] the error appears in the field itself, rather than as a tool tip
- [x] the error message is persistent if the user tabs beyond that field
- [ ] screen reader users hear "error" before the error message itself

- [ ] Appropriate cypress/unit tests are in place

**Comment by mmiddaugh** [comment]
Please refer to [recording with Josh](https://drive.google.com/file/d/1drsmPly33CQHGzVsxrvY_0y9OmMiHrZR/view?usp=sharing)

**Comment by mmiddaugh** [comment]
Per Chris Valarida, the current forms library is not a good option for the Facility Locator as it is really intended for the digitization of paper forms[Slack thread here](https://dsva.slack.com/archives/G01PBG4D397/p1614030445000100)
Any improvements to the current forms library likely won't be available for early adoption until probably Q3 at the earliest.

**Comment by mmiddaugh** [comment]
Scenario: If the user interacts with the map before completing fields, the "Search this area" button becomes visible. 

Decision: Error message handling for "Search this area" button will be the same as for the "Search" button. 
- "Search this area" button will be visible before all fields are completed (same as "Search" button). 
- Errors are announced if user clicks either search function before all required fields are populated. 
- Form fields are not errored when use arrives on page.

**Comment by mmiddaugh** [comment]
@joshkimux we've love your feedback when you are able

**Comment by joshkimux** [comment]
@mmiddaugh I'll give this a look this week now that I'm back from PTO! 👍 🙇

**Comment by joshkimux** [comment]
@mmiddaugh I've reviewed this using Safari and VO and it's a fantastic improvement. Great work team! 🥳 ✨ 

One last thing before we close this out-- **could we add in a `sr-only` span with "error" written inside it ahead of the error message itself?** This will announce to screen reader users that what they are hearing is indeed an error and not just a label which is important to indicating that a change must be made. For example:

```html
<span class="sr-only">Error</span>
```

This will also ensure the [facility locator error messages meet design system standards](https://design.va.gov/patterns/form-feedback) 😄

**Comment by joshkimux** [comment]
@mmiddaugh If y'all have some extra time, I'd also recommend exploring alternative methods to placeholder text. Veterans may forget what the examples were upon clicking into the field, which may become a memory challenge (see video below). [These concerns are officially documented in our design system here.](https://design.va.gov/components/form-controls#accessibility)

Ideally, hint text should live above the field and below the label. [There are conversations right now on adding this to the design system.](https://app.mural.co/t/vsa8243/m/vsa8243/1613766593541/3c07de4000524ac050d6296afe88daed177ec5ad?wid=0-1613766656344)

For more info on why this works, [I recommend checking out this article by Adam Silver on placeholder text!](https://adamsilver.io/articles/placeholders-are-problematic/)

https://user-images.githubusercontent.com/14154792/110373356-67dee580-801d-11eb-9d70-2d21207f557c.mov

**Comment by mmiddaugh** [comment]
Following up on Josh's comments above - 
1. add in a sr-only span with "error" written inside it ahead of the error message itself -> we will groom this addition this afternoon and finish out the work as you suggest
2. Explore alternative methods to placeholder text -> I will create an issue for this suggestion as an iteration.

**Comment by mmiddaugh** [comment]
Tracking remaining work in #21179 so this can be closed
